include/swift/Runtime/Metadata.h

```diff
@@ -4389,6 +4389,12 @@ void swift_getFieldAt(
     const Metadata *type, unsigned index,
     std::function<void(llvm::StringRef name, FieldType type)> callback);
 
+#if !NDEBUG
+/// Verify that the given metadata pointer correctly roundtrips its
+/// mangled name through the demangler.
+void verifyMangledNameRoundtrip(const Metadata *metadata);
+#endif
```

Comment:
Maybe when `NDEBUG`, we should add 

```
  void verifyMangledNameRoundtrip(const Metadata *metadata) { }
```

so the call sites don't need to check `NDEBUG` themselves?

Reply:
There's currently just one call site, and I like that it's explicit on the caller's side that the check doesn't happen when `NDEBUG` is set. However, I am open to be persuaded otherwise.